### PR TITLE
Remove old releases from target host after deployment

### PR DESCRIPTION
**Pull request by wladekb**
This should solve our problems with disk space on target machine as all the releases except the current one will be automatically removed after deployment is finished.

/cc @harnash @jcellary

### CODE REVIEW

File: apps/testrunner/tasks/deploy.py [hunk 1]
```diff
@@ -2,22 +2,32 @@
...
 
     @classmethod
     def run_prep(cls, connection, app, env, repos):
         repo_spec = ' '.join([
-            '-r {repo_name}@{repo_commit}'.format(repo_name=repo_name, repo_commit=repo_commit)
-            for repo_name, repo_commit in repos.items()
-        ])
+                                 '-r {repo_name}@{repo_commit}'.format(repo_name=repo_name, repo_commit=repo_commit)
+                                 for repo_name, repo_commit in repos.items()
+                                 ])
```
Comment (harnash) [inline apps/testrunner/tasks/deploy.py]: Here indentation looks weird....

File: apps/testrunner/tasks/deploy.py [hunk 2]
```diff
@@ -36,13 +46,45 @@ def run_push(cls, connection, app, env):
 
         cls.run_remote_command(connection, cmd)
 
+    @classmethod
+    def clean_up_old_releases(cls, hostname, connection):
+        current_release = int(cls.get_remote_output(connectoin, 'readlink {}current'.format(cls.DEPLOY_DIR)))
```
Comment (harnash) [inline apps/testrunner/tasks/deploy.py]: shouldn't we check if conversion was successful?

Comment (harnash) [inline apps/testrunner/tasks/deploy.py]: ```connectoin``` -> ```connection```

File: apps/testrunner/tasks/deploy.py [hunk 3]
```diff
@@ -36,13 +46,45 @@ def run_push(cls, connection, app, env):
...
+        logger.debug('{}: current_release = {}'.format(hostname, current_release))
+
+        deployed_releases = cls.get_remote_output(connection, 'ls {}'.format(cls.DEPLOY_DIR)).split()
+        logger.debug('{}: deployed_releases = {}'.format(hostname, deployed_releases))
+
+        for f in deployed_releases:
+            if f == 'current' or f == current_release_str or f.startswith('.'):
+                continue
+            logger.debug('{}: removing deployed release {}'.format(hostname, f))
+            cls.run_remote_command(connection, 'rm -rf {}{}'.format(cls.DEPLOY_DIR, f))
```
Comment (harnash) [inline apps/testrunner/tasks/deploy.py]: We should really come up with better solution for this since it is a major security hole

**Comment by harnash** [comment]
For now LGTM (except above remarks) but we should come up with more elegant solution.

**Comment by harnash** [comment]
Oh. Please do merge with master before merging this PR to see if Travis-ci tests will pass :)

File: apps/testrunner/tasks/deploy.py (hunk 2 above)
Comment (wladekb) [inline apps/testrunner/tasks/deploy.py]: unsuccessful conversion will trigger exception and we'll know about it

**Pull request merged**